### ISSUE TITLE
how to avoid black patches when zoom in and zoom out

### ISSUE THREAD

**Issue by sssvrock** [barteksc/AndroidPdfViewer#263]
When zooming the view it is showing the black patches and it is not yet good when doing this operations. Please suggest me how can i fix this

**Comment by 1stmetro** [comment]
Black patches? can you post the PDF and or an image showing your issue?

**Comment by sssvrock** [comment]
Please find the black patches when zooming the pdf files

![1](https://cloud.githubusercontent.com/assets/1411543/24896899/8d99359c-1eb4-11e7-8841-e680eea17cc2.png)
![2](https://cloud.githubusercontent.com/assets/1411543/24896901/8e081610-1eb4-11e7-8923-90c238d3cad2.png)
![3](https://cloud.githubusercontent.com/assets/1411543/24896900/8e0506f0-1eb4-11e7-9e5b-7adfd0cf3a6e.png)

**Comment by 1stmetro** [comment]
Change the constants.settings it will improve this issue and maybe fix it too

**Comment by SagarAghara** [comment]
hey @sssvrock ..did you solved your Problem of Black Patches...??..can you help me if you solved..

**Comment by 1stmetro** [comment]
Just tweek the constants to you have the desired results.

**Comment by SagarAghara** [comment]
@1stmetro ..thanks for your valuable comment....but i didn't get you...plz tell me how can i do as you told.....constants...????what

**Comment by sssvrock** [comment]
Hi,
  Instead of gradle plugin, I downloaded the source code of library and made changes in Constants.
It is working fine.

> By deault it is : 0.3f
> public static float THUMBNAIL_RATIO = 0.7f;

**Comment by SagarAghara** [comment]
Ok. Thanks...Means i have to Copy & Paste all libs class from this Library. and in Constant class change only this Variable value THUMBNAIL_RATIO.Thanks i am trying.

**Comment by barteksc** [comment]
This field is `public static`, why you are downloading sources?

**Comment by SagarAghara** [comment]
Yeah sry Bro. i forget it..Thanks

**Comment by SagarAghara** [comment]
but without copy paste of your libs code. how can i edit your THUMBNAIL_RATIO value..bcz we can not edit  Libs. i read it from Internet.you use THUMBNAIL_RATIO variable in PagesLoader class on loadPages().

**Comment by KoTius** [comment]
@SagarAghara, @sssvrock you don't need to edit the lib. It is PUBLIC static constant. Its available from everywhere in your project. Just change it.
com.github.barteksc.pdfviewer.util.Constants.THUMBNAIL_RATIO = 0.7f;

You can see this class with your eyes, there are few usefull constains you can change aswell, Constants.MAXIMUM_ZOOM, for example

**Comment by SagarAghara** [comment]
@KoTius Thanks for your ans....i tried it more........well see below is my class where i used PDFViewer for view PDF Files...please tell me when you will be free..

**PDFViewActivity.class**

public class PDFViewActivity extends AppCompatActivity {
    PDFView pdfView;
    String pdfFileName, PDFPath, SharePath;
    Integer pageNumber = 0;
    File PDFFile;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        GlobalConstant.ActivityFullScreen(this);
        setContentView(R.layout.activity_pdfview);
        
        pdfView = (PDFView) findViewById(R.id.pdfView);    
        pdfFileName = getIntent().getExtras().getString("pdfFileName");
        
        PDFPath = Environment.getExternalStorageDirectory() + "/PDFs/" + pdfFileName;
        PDFFile = new File(PDFPath);
        displayPDF();
    }
    private void displayPDF() {
        pdfView.fromFile(PDFFile)
                .defaultPage(pageNumber)
                .enableSwipe(true)
                .swipeHorizontal(false)
                .enableAnnotationRendering(true)
                .scrollHandle(new DefaultScrollHandle(this))
                .load();
        pdfView.useBestQuality(true);
    }
 }

see.. i have fetched **pdfFileName**  from another Activity and Display that particular PDF in this lib.
all are ok except Black Patches....Help me out from this Problem.....Please tell me where i put below line 
**com.github.barteksc.pdfviewer.util.Constants.THUMBNAIL_RATIO = 0.7f;**

**Comment by KoTius** [comment]
@SagarAghara it's just global variable. Change it wherever you want. You can do it in OnCreate().

**Comment by SagarAghara** [comment]
i put it on onCreate()....Thanks bro. God Bless You...but Black patches still Remaining...):

**Comment by KoTius** [comment]
@SagarAghara well, it worked for me. Dunno, is there are any chance, that that variable changes by another source. You can try to put it right before pdfView.fromFile() and see if there are difference.

**Comment by SagarAghara** [comment]
yes bro. i put it before calling my  displayPDF().........means before  pdfView.fromFile().

**Comment by SagarAghara** [comment]
In my App i generated PDF programmatically..and my PDF file contains Tables.....i think...because of table black patches displayed...It seems after putting this line com.github.barteksc.pdfviewer.util.Constants.THUMBNAIL_RATIO = 0.7f;
some patches remove...now some patches is showing....now that's enough Thanks for your Help.

**Comment by argmnt** [comment]
--**in pdfviewer.util;
--in Constants.java** file

changing line below
**public static float PART_SIZE = 250;**
to:
**public static float PART_SIZE = 600;**
worked for me!

I did not dig that much about this static field but since this is a static field, according to your pdf size resolution, you can change value of this static field before rendering PdfView !!!

--------------------------------

**While setting this field rendering waiting time, below part shouldn't be forgetten;  As doc says:**
/**
     * The size of the rendered parts (default 256)
     * Tinier : a little bit slower to have the whole page rendered but more reactive.
     * Bigger : user will have to wait longer to have the first visual results
     */

**Change will effect to waiting time.** 
So What you can do is adjusting value of PART_SIZE field but for better waiting rendering time you can make a trade-off between MAXIMUM_ZOOM and PART_SIZE fields. For example; You can say this MAXIMUM_ZOOM value suits my app requirements then according to that zoom value, You can set minumum value of PART_SIZE field!

Fields:
**Constants.java**
public static float PART_SIZE 
public static float MAXIMUM_ZOOM = 10;
